src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -149,6 +153,8 @@ internal partial class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
 
         public RpcWorkerConfig WorkerConfig => _workerConfig;
 
+        public IOptions<ScriptJobHostOptions> JobHostOptions => (IOptions<ScriptJobHostOptions>)_serviceProvider.GetService(typeof(IOptions<ScriptJobHostOptions>));
```

Comment:
Can we fetch this only once and store the value?

Reply:
We can cache but need to take in to account specialization where this value will/could change

Reply:
@jviau I've refactored this to use ActiveHostChanged event for that scenario, thoughts?

https://github.com/Azure/azure-functions-host/pull/9523/commits/263a5b6e920e865235b0cc7a4b643387e0a49dc2